iphone/Maps/Classes/Editor/Street/MWMStreetEditorViewController.mm

```diff
@@ -45,15 +47,16 @@ - (void)configNavBar
 
 - (void)configData
 {
-  self.streets = [[self.delegate getNearbyStreets] mutableCopy];
-  NSString * currentStreet = [self.delegate getStreet];
-  BOOL const haveCurrentStreet = (currentStreet && currentStreet.length != 0);
+  m_streets = self.delegate.nearbyStreets;
+  auto const & currentStreet = self.delegate.currentStreet;
+
+  BOOL const haveCurrentStreet = !currentStreet.m_defaultName.empty();
   if (haveCurrentStreet)
   {
-    [self.streets removeObject:currentStreet];
-    [self.streets insertObject:currentStreet atIndex:0];
+    m_streets.erase(remove(m_streets.begin(), m_streets.end(), currentStreet));
```

Comment:
OK, I see why do you need vector here. Could you please explain why do you need to move currentStreet to the top of the list?